crates/sui-graphql-rpc/src/extensions/logger.rs

```diff
@@ -0,0 +1,150 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use async_graphql::{
+    extensions::{
+        Extension, ExtensionContext, ExtensionFactory, NextExecute, NextParseQuery, NextRequest,
+        NextValidation,
+    },
+    parser::types::{ExecutableDocument, OperationType, Selection},
+    PathSegment, Response, ServerError, ServerResult, ValidationResult, Variables,
+};
+use futures_util::lock::Mutex;
+use std::{fmt::Write, sync::Arc};
+use tracing::{error, info};
+use uuid::Uuid;
+
+// TODO: mode in-depth logging to debug
+
+#[derive(Clone, Debug)]
+pub struct LoggerConfig {
+    pub log_request_query: bool,
+    pub log_response: bool,
+    pub log_complexity: bool,
+}
+
+impl Default for LoggerConfig {
+    fn default() -> Self {
+        Self {
+            log_request_query: true,
+            log_response: true,
+            log_complexity: true,
+        }
+    }
+}
+
+#[derive(Clone, Debug, Default)]
+pub struct Logger {
+    config: LoggerConfig,
+}
+
+impl ExtensionFactory for Logger {
+    fn create(&self) -> Arc<dyn Extension> {
+        Arc::new(LoggerExtension {
+            session_id: None.into(),
+            config: self.config.clone(),
+        })
+    }
+}
+
+struct LoggerExtension {
+    session_id: Mutex<Option<Uuid>>,
```

Comment:
Not really experienced with tokio and futures, but couldn't tokio Mutex be used here?

Reply:
generally you actually want to avoid using tokio Mutex and prefer the stdlib's

Reply:
> generally you actually want to avoid using tokio Mutex and prefer the stdlib's

but in async context, we dont want std Mutex to block, no?

Reply:
>but in async context, we dont want std Mutex to block, no?

This isn't actually an issue actually, you can read the link i put above for more on this. Its actually preferable to use stdlib locks everywhere if possible